File: Excel/900-999/982/Challenge 982.py
import pandas as pd

path = "900-999/982/982 Palindromic Creation.xlsx"
input = pd.read_excel(path, usecols="A", nrows=8, skiprows=0)
test = pd.read_excel(path, usecols="B", nrows=8, skiprows=0)

input["out"] = input.Words.map(lambda s: next(s+s[:i][::-1] for i in range(len(s)) if s[i:]==s[i:][::-1]))

print(input['out'].equals(test['Answer Expected']))
# True

File: Excel/900-999/982/Challenge 982.R
library(tidyverse)
library(readxl)

path <- "900-999/982/982 Palindromic Creation.xlsx"
input <- read_excel(path, range = "A1:A8")
test <- read_excel(path, range = "B1:B8")

result = input %>%
  mutate(
    out = map_chr(Words, \(s) {
      detect(seq_along(str_split_1(s, "")) - 1, \(i) {
        str_sub(s, i + 1) == str_rev(str_sub(s, i + 1))
      }) |>
        {
          \(i) str_c(s, str_rev(str_sub(s, 1, i)))
        }()
    })
  )

all.equal(result$out, test$`Answer Expected`)
# [1] TRUE
